crates/uv/src/commands/project/run.rs

```diff
@@ -742,6 +741,20 @@ impl RunCommand {
                 process
             }
             Self::External(executable, args) => {
+                // If the executable is `python` and it's not on the path, rename it to the
+                // path reported by the interpreter.
```

Comment:
Could this not resolve to some Python that doesn't match the intended interpreter? Like you run `uv run -p 3.9`, but we then pick up Python 3.12 that's on your path?

Reply:
Yeah but what if you have a `python` executable in the interpreter bin that does some special behavior? Should we unconditionally switch to the sys executable? Should we just look in the interpreter bin instead of the whole PATH here?

Reply:
Seems problematic both ways. I'm okay with unconditional replacement, I guess?

Reply:
Here's that approach https://github.com/astral-sh/uv/pull/6363